Code/Core/FileIO/FileIO.cpp

```diff
@@ -898,12 +898,12 @@
         if ( readOnly )
         {
             // remove writable flag
-            s.st_mode &= ( ~S_IWUSR ); // TODO:LINUX Is this the correct behaviour?
+            s.st_mode &= ~( S_IWUSR | S_IWGRP | S_IWOTH );
```

Comment:
If setting writeable only adds the user flag, it seems that removing writable should also only remove thwe user writable flag?